packages/synthetix-main/contracts/satellites/AccountToken.sol

```diff
@@ -47,12 +47,21 @@ contract AccountToken is IAccountToken, ERC721, AccountTokenStorage, Initializab
     // ---------------------------------------
     // Mint/Transfer
     // ---------------------------------------
-    function mint(address owner, uint256 accountId) external override {
-        _mint(owner, accountId);
+    function mint(address accountOwner, uint256 accountId) public override {
+        _mint(accountOwner, accountId);
 
-        IAccountModule(_accountStore().mainProxy).transferAccount(owner, accountId);
+        IAccountModule(_accountStore().mainProxy).transferAccount(accountOwner, accountId);
 
-        emit AccountMinted(owner, accountId);
+        emit AccountMinted(accountOwner, accountId);
+    }
+
+    function mintNext(address accountOwner) external override returns (uint256 accountId) {
+        while (_exists(_accountStore().recentIdUsed)) {
```

Comment:
you can check before if the number of tokens created is less than recentIdUsed and fast forward to it, also write on the while loop to an internal variable and at the end go to the store since it's cheaper. 
i.e.

```uint checkId =  _accountStore().recentIdUsed;
if (checkId  < _accountStore().balance) {
   checkId = _accountStore().balance;
}
while (_exists(checkId) {
  checkId++;
}
accountId = checkId;
_accountStore().recentIdUsed = accountId;
```
and you need to add the balance in the store (`uint256 balance,`) and an increment to the public mint function (`_accountStore().balance++;`)